File: Leetcode/611_Leetcode.py
from typing import List

class Solution:
    def countTriangles(self, nums: List[int]):
        res = 0
        nums.sort()

        for i in range (2, len(nums)):
            left, right = 0, i - 1

            while left < right:
                if nums[left] + nums[right] > nums[i]:
                    res += right - left
                    right -= 1
                else:
                    left += 1
        return res

File: Leetcode/611_Leetcode.js
function countTriangles(nums) {
    let res = 0;
    nums.sort((a, b) => a - b);

    for (i = 2; i < nums.length; ++i){
        let left = 0; right = i - 1;
        
        while (left < right){
            if (nums[left] + nums[right] > nums[i]){
                res += right - left;
                right--;
            }
            else{
                left++;
            }
        }
    }
    return res;
    
}
const arr = [4, 6, 3, 7];
console.log(countTriangles(arr));